### PR TITLE
add `dvc-checkpoints-mnist`

### PR DESCRIPTION
**Pull request by casperdcl**
- Generate https://github.com/iterative/dvc-checkpoints-mnist

### CODE REVIEW

File: dvc-checkpoints-mnist/branches/full_pipeline/train.py [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+    x = dataset.data.reshape(len(dataset.data), 1, 28, 28) / 255
+    y = dataset.targets
+    return x, y
+
+
+def train(model, x, y, lr, weight_decay):
+    """Train a single epoch."""
+    model.train()
+    criterion = torch.nn.CrossEntropyLoss()
+    optimizer = torch.optim.Adam(model.parameters(), lr=lr, weight_decay=weight_decay)
```
Comment (casperdcl) [inline dvc-checkpoints-mnist/branches/full_pipeline/train.py:46]: @dberenbaum isn't this missing `optimizer.load_state_dict()` to ensure the epoch number & weight decay are used correctly?

**Comment by shcheklein** [comment]
Question, @casperdcl @iesahin @dberenbaum - how can we consolidate the repos?

At the moment I'm completely lost in all those repositories. We need to cleanup, come up with the name convention, reuse one or two repos to do experiments, or at least reuse the same code base.

@iesahin what is current status of the experiments-get-started? does it cover checkpoints? what is missing there?

**Comment by dberenbaum** [comment]
My current understanding:

- https://github.com/iterative/dvc-checkpoints-mnist.git is referenced in https://dvc.org/doc/dvclive/user-guide/dvclive-with-dvc and nowhere else. I think it's worth fixing small issues in that repo since it is still being referenced.
- @flippedcoder maintains https://github.com/iterative/checkpoints-tutorial, which is referenced in https://dvc.org/doc/user-guide/experiment-management/checkpoints and nowhere else. I think it's a fork of the first repo, and I'm not sure of the differences. It's also probably worth making updates to this repo and corresponding changes to that tutorial as long as needed.
- @iesahin is working on https://github.com/iterative/get-started-experiments. I hope that this repo or a similar one can eventually be the source for checkpoints docs.

I'm not sure how close we are to switching to the repo of @iesahin. Ideally, I would prefer not to take the time to auto generate the other repos if we think they can be replaced soon.

Sorry @casperdcl if this was unclear. I didn't realize you were planning to try to generate this repo. Maybe we need it if we aren't close with the new repo, but I hope not.

**Comment by iesahin** [comment]
If we have decided on merging `dvc-checkpoints-mnist` and `get-started-checkpoints` and use the latter for all the checkpoints related projects, I can synchronize `get-started-checkpoints` with `get-started-experiments`. `get-started-checkpoints` is already codified in https://github.com/iterative/example-repos-dev/pull/45. It needs some update but I believe it will be quicker than adding a new codification for `dvc-checkpoints-mnist`.

**Comment by iesahin** [comment]
> We need to cleanup, come up with the name convention, reuse one or two repos to do experiments, or at least reuse the same code base.

We visited these topics in the past, e.g. in https://github.com/iterative/example-repos-dev/issues/34. 

- Simple repositories and a single code base seems to contradict at some level. I could use https://github.com/iterative/dvc-get-started for all the experiments / checkpoints / pipelines stuff, but it began to look too complex. I tend to keep per-feature repositories, and they evolve to different code bases. For example, `get-started-pipelines` and `get-started-experiments` started with the same code base but now they are very different. 
- Standardizing on the names may be good. I can update the names [according to the card](https://github.com/orgs/iterative/projects/172#card-67676327) in this week's project: `get-started-experiments` ➡️ `dvc-example-experiments-tensorflow`, `get-started-pipelines` ➡️  `dvc-example-pipelines-tensorflow` and `get-started-checkpoints` ➡️ `dvc-example-checkpoints-tensorflow`. Additionally `dvc-checkpoints-mnist` can be `dvc-example-checkpoints-torch`, `checkpoints-tutorial` can be `dvc-example-checkpoints-torch-tutorial`. `example-get-started` is referenced in too many places, so I believe it's better to keep it intact.

**Comment by dberenbaum** [comment]
@iesahin I'm fine with using `get-started-checkpoints` for future checkpoints-related docs. Do you see anything in `dvc-checkpoints-mnist` that you think needs to be merged or added in `get-started-checkpoints`? I don't think it's necessary to keep anything from `dvc-checkpoints-mnist` unless it's missing from the other repos.

**Comment by iesahin** [comment]
` get-started-experiments`  seems to diverge quite a bit from `get-started-checkpoints`. Instead of modifying the latter, I'm updating `get-started-experiments` to get a new repository `dvc-example-checkpoints-tensorflow`.

**Comment by iesahin** [comment]
Do we need this particular PR to codify `dvc-checkpoints-mnist`? Could we close this? @casperdcl @dberenbaum

**Comment by dberenbaum** [comment]
IMO we don't need to codify `dvc-checkpoints-mnist`.

**Comment by casperdcl** [comment]
I'm not tracking this any more; feel free to take over @iesahin or close it.
